sourcegraph/sourcegraph
# JetBrains: Cody: Allow manual autocomplete to be triggered even when implicit completions are disabled

Fixes #56345 

- manual autocomplete should now be able to bypass implicit autocomplete configuration
- some util classes were cleaned up, moved or translated to Kotlin
- some dead code was removed

## Test plan
- check if manual autocomplete (alt + backslash) can be triggered when implicit autocomplete is disabled in settings
- check if manual autocomplete can be triggered when implicit autocomplete is disabled for a particular language in settings
- check if autocomplete inlays are cleared correctly in the scenarios mentioned above
- check if other autocomplete uses work as expected
